torchgeo/datasets/satlas.py

```diff
@@ -0,0 +1,700 @@
+        'colors': [
+            [0, 0, 0],  # (black) invalid
+            [0, 255, 0],  # (green) background
+            [0, 0, 255],  # (blue) water_event
+        ],
+    },
+    'park_sport': {
+        'type': 'classification',
+        'categories': [
+            'american_football',
+            'badminton',
+            'baseball',
+            'basketball',
+            'cricket',
+            'rugby',
+            'soccer',
+            'tennis',
+            'volleyball',
+        ],
+    },
+    'park_type': {
+        'type': 'classification',
+        'categories': ['park', 'pitch', 'golf_course', 'cemetery'],
+    },
+    'power_plant_type': {
+        'type': 'classification',
+        'categories': ['oil', 'nuclear', 'coal', 'gas'],
+    },
+    'quarry_resource': {
+        'type': 'classification',
+        'categories': ['sand', 'gravel', 'clay', 'coal', 'peat'],
+    },
+    'track_sport': {
+        'type': 'classification',
+        'categories': ['running', 'cycling', 'horse'],
+    },
+    'road_type': {
+        'type': 'classification',
+        'categories': [
+            'motorway',
+            'trunk',
+            'primary',
+            'secondary',
+            'tertiary',
+            'residential',
+            'service',
+            'track',
+            'pedestrian',
+        ],
+    },
+    'cloud': {
+        'type': 'bin_segment',
+        'categories': ['background', 'cloud', 'shadow'],
+        'colors': [
+            [0, 255, 0],  # (green) not clouds or shadows
+            [255, 255, 255],  # (white) clouds
+            [128, 128, 128],  # (grey) shadows
+        ],
+        'BackgroundInvalid': True,
+    },
+    'flood': {
+        'type': 'bin_segment',
+        'categories': ['background', 'water'],
+        'colors': [
+            [0, 255, 0],  # (green) background
+            [0, 0, 255],  # (blue) water
+        ],
+        'BackgroundInvalid': True,
+    },
+}
+
+
+class SatlasPretrain(NonGeoDataset):
+    """SatlasPretrain dataset.
+
+    `SatlasPretrain <https://satlas-pretrain.allen.ai/>`_ is a large-scale pre-training
+    dataset for tasks that involve understanding satellite images. Regularly-updated
+    satellite data is publicly available for much of the Earth through sources such as
+    Sentinel-2 and NAIP, and can inform numerous applications from tackling illegal
+    deforestation to monitoring marine infrastructure. However, developing automatic
+    computer vision systems to parse these images requires a huge amount of manual
+    labeling of training data. By combining over 30 TB of satellite images with 137
+    label categories, SatlasPretrain serves as an effective pre-training dataset that
+    greatly reduces the effort needed to develop robust models for downstream satellite
+    image applications.
+
+    Reference implementation:
+
+    * https://github.com/allenai/satlas/blob/main/satlas/model/dataset.py
```

Comment:
Not sure if it's worth adding this in the docstring or just leaving it in the comments only. Happy to add pointers to the official codebase/data loaders for Satlas which may be preferred by some users.